### PR TITLE
Add `SUBSTRING` to non-reserved SQL keywords

### PR DESCRIPTION
**Pull request by seut**
## Summary of the changes / Why this improves CrateDB

This enables support for using SUBSTRING with the generic function call syntax and improves PostgreSQL compatibility.
Example:
```SELECT substring('crate', 1, 3);```

Although this syntax for SUBSTRING is not official documented neither at PostgreSQL nor by the SQL standard, it is indeed
supported by PostgreSQL and used by applications like e.g. Metabase (see #12620).

Note that this differs from the SQL standard wheras `SUBSTRING` is defined as a reserved keyword.

Relates #12620.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -116,6 +116,10 @@ Changes
 
 - Updated Admin UI to 1.22.0, including an update with the new logo and colors.
 
+- Added ``SUBSTRING`` to non-reserved SQL keywords in order to support the
```
Comment (matriv) [inline docs/appendices/release-notes/unreleased.rst:119]: Please check indentation, seems that also the above entry is not consistent.

I guess this is not considered breaking, only the other way around (non-reserved->reserved), right?

**Review by matriv — APPROVED** [review]
LGTM. thx! left a comment on the changelog.

File: docs/appendices/release-notes/unreleased.rst (hunk 1 above)
Comment (seut) [reply docs/appendices/release-notes/unreleased.rst:119]: > Please check indentation, seems that also the above entry is not consistent.

Hm looks fine to me. What's wrong here?

> I guess this is not considered breaking, only the other way around (non-reserved->reserved), right?

I'd say so as being more lenient should not break anything. Or do I miss something?

**Comment by matriv** [comment]
@seut Unless I'm missing something, I think it's not enough to change the parser only. Don't we need to register the function?
I tried with the branch and I get:
```
crate=> select substring('marios' FROM 4);
 'ios' 
-------
 ios
(1 row)

crate=> select substring('marios', 1);
ERROR:  Unknown function: substring('marios', 1)
```

**Comment by seut** [comment]
@matriv pushed a fixup

File: docs/appendices/release-notes/unreleased.rst (hunk 1 above)
Comment (matriv) [reply docs/appendices/release-notes/unreleased.rst:119]: Sorry, the github view shows it strangely as it's indented more:
Screenshot_20220617_120955

**Review by matriv — APPROVED** [review]
thx!

**Pull request merged**